### PR TITLE
Strip schema from table for Join::foreign_field guess

### PR DESCRIPTION
**Pull request by mvorisek**
related with https://github.com/atk4/data/pull/1002, at least Join probably needs a simillar fix

/cc @samsouthardjr

### CODE REVIEW

**Comment by samsouthardjr** [comment]
> related with #1002, at least Join probably needs a simillar fix
> 
> /cc @samsouthardjr

I'm not an expert at Join or github, but you're right.  The exact same fix should be made at line 357 of Join.php.  Are you going to put that into this branch, or were you looking for me to do so?  I've never contributed to a pull request that someone else has started, but if you're looking to me to do it I can always try and learn.

**Comment by mvorisek** [comment]
Technically you can open a PR into `join_name_guess_from_table_with_schema` branch, but I fixed it - the only thing to fix is the test and you can offer (such small) fix using a GH suggestion:

image

**Comment by samsouthardjr** [comment]
> Technically you can open a PR into `join_name_guess_from_table_with_schema` branch, but I fixed it - the only thing to fix is the test and you can offer (such small) fix using a GH suggestion:
> 
> image

I couldn't find that button -- is it in githubdesktop, maybe?

In any case, I'll just comment on it here, since it's your pull.  The current pull doesn't address that the same change is needed in Join's hasMany() method, at line 357.

File: src/Model/Join.php [hunk 1]
```diff
@@ -238,7 +238,7 @@ protected function init(): void
             }
 
             if (!$this->foreign_field) {
-                $this->foreign_field = $this->getModelTableString($this->getOwner()) . '_' . $id_field;
+                $this->foreign_field = preg_replace('~^.+?\.~', '', $this->getModelTableString($this->getOwner())) . '_' . $id_field;
```
Comment (samsouthardjr) [inline src/Model/Join.php:243]: Needs the same change on line 357 of this file

Comment (mvorisek) [reply src/Model/Join.php:243]: yes! - fixed, I think the seeding on l357 was just a duplicated/unneeded code, so I removed it

Comment (mvorisek) [reply src/Model/Join.php:243]: > I couldn't find that button -- is it in githubdesktop, maybe?

available only in "Files changed" tab and near the changed lines only

**Comment by mvorisek** [comment]
Impl. fixed, but Join should be dropped sooner or later with Model impl. to support full/nested modelling.

**Pull request merged**